### PR TITLE
File: support previous source maps

### PR DESCRIPTION
**Pull request by miripiruni**
Fix for https://github.com/enb/enb-bemxjst/issues/283

Relates: 

 * https://github.com/bem/bem-xjst/pull/491
 * https://github.com/enb/enb-bemxjst/pull/284

### CODE REVIEW

**Comment by miripiruni** [comment]
cc @sbmaxx

**Comment by sbmaxx** [comment]
lgtm

**Comment by miripiruni** [comment]
@tadatuta @zxqfox review needed

**Review by qfox — APPROVED** [review]

File: lib/file.js [hunk 1]
```diff
@@ -22,11 +23,15 @@ function File(filename, opts) {
     opts.comment = opts.comment || 'inline';
 
     this._content = [''];
+
     if (opts.sourceMap) {
         this._map = new SourceMapGenerator({
-            file: path.basename(filename),
+            file: opts.sourceMap.from,
```
Comment (tadatuta) [inline lib/file.js]: according to JSDoc `sourceMap` should be boolean.
and test for such case is also missing.

And BTW the change looks major. To make it minor let's keep fallback:

```js
opts.sourceMap.from || path.basename(filename)
```

Comment (miripiruni) [reply lib/file.js]: @tadatuta 👌

**Pull request merged**